test/cpp/api/functional.cpp

```diff
@@ -843,6 +843,20 @@ TEST_F(FunctionalTest, CrossEntropy) {
   ASSERT_TRUE(F::cross_entropy(input, target).allclose(expected, 1e-04));
 }
 
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+TEST_F(FunctionalTest, CrossEntropyWithSoftLabels) {
+  auto input = torch::tensor({{3., 3.}, {2., 2.}}, torch::kFloat);
+  auto target = torch::tensor({{1., 0.}, {0., 1.}}, torch::kFloat);
+  auto output = F::cross_entropy_with_soft_labels(
+      input, target, F::CrossEntropyWithSoftLabelsFuncOptions().reduction(torch::kMean));
+  // Using one-hot labels should be equivalent to cross entropy loss except
+  // smaller by a factor of num classes (2 in this case) for the mean reduction.
+  auto expected = torch::tensor(0.6931 / 2., torch::kFloat);
```

Comment:
nit: It says "Using one-hot labels should be equivalent to cross entropy loss except" but then we hardcode in "torch::tensor(0.6931 / 2., torch::kFloat)", would it be better if we called the c++ version of F.cross_entropy and then used that as the `expected` tensor?

Ditto in the module test

Reply:
Cool, I like that better